The build fails on i386 systems with no __SSE__ due to missing _mm_pause()

Hi!

On the mageia i586 build system I am getting this:

```
 /home/iurt/rpmbuild/BUILD/opencv-3.4.1/modules/core/src/parallel_impl.cpp:396:21: note: in expansion of macro 'CV_PAUSE' CV_PAUSE(16); ^~~~~~~~ /home/iurt/rpmbuild/BUILD/opencv-3.4.1/modules/core/src/parallel_impl.cpp: In member function 'void cv::ThreadPool::run(const cv::Range&, const cv::ParallelLoopBody&, double)': /home/iurt/rpmbuild/BUILD/opencv-3.4.1/modules/core/src/parallel_impl.cpp:53:79: error: '_mm_pause' was not declared in this scope # define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { _mm_pause(); } } while (0) ^ /home/iurt/rpmbuild/BUILD/opencv-3.4.1/modules/core/src/parallel_impl.cpp:673:33: note: in expansion of macro 'CV_PAUSE' CV_PAUSE(16);
```

Please look into fixing it.

--- a/modules/core/src/parallel_impl.cpp
+++ b/modules/core/src/parallel_impl.cpp
@@ -49,7 +49,11 @@ DECLARE_CV_YIELD
 DECLARE_CV_PAUSE
 #endif
 #ifndef CV_PAUSE
-#if defined __GNUC__ && (defined __i386__ || defined __x86_64__)
+# if defined __GNUC__ && (defined __i386__ || defined __x86_64__)
+#   if !defined(__SSE__)
+      static inline void cv_non_sse_mm_pause() { __asm__ __volatile__ ("rep; nop"); }
+#     define _mm_pause cv_non_sse_mm_pause
+#   endif
 #   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { _mm_pause(); } } while (0)
 # elif defined __GNUC__ && defined __aarch64__
 #   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { asm volatile("yield" ::: "memory"); } } while (0)

